test/common/network/lc_trie_test.cc

```diff
@@ -32,17 +32,23 @@ class LcTrieTest : public testing::Test {
     }
   }
 
-  void expectIPAndTag(const std::vector<std::pair<std::string, std::string>>& test_output) {
+  void
+  expectIPAndTag(const std::vector<std::pair<std::string, std::vector<std::string>>>& test_output) {
     for (const auto& kv : test_output) {
-      EXPECT_EQ(kv.second, trie_->getTag(Utility::parseInternetAddress(kv.first)));
+      EXPECT_EQ(kv.second, trie_->getTags(Utility::parseInternetAddress(kv.first)));
+    }
+  }
+
+  void expectIPAndNoTag(const std::vector<std::string>& test_output) {
+    for (const auto& ip : test_output) {
+      EXPECT_EQ(0, trie_->getTags(Utility::parseInternetAddress(ip)).size());
```

Comment:
My only thought on the test refactor is that if multi-vector responses are coming, you may want to refactor expectIPAndTag to actually take a vector, rather than splitting this into "one string" "empty vector" and "multi-string response" test cases.  Your call if it's easier this way.

Reply:
I got rid of expectIPandNoTag() and kept ExpectIPandTags(). It checks the content of the tag vector against the expected vector, empty vectors are supported too!